### PR TITLE
DEV: removes jquery usage from highlight-syntax

### PR DESCRIPTION
**Pull request by jjaffeux**
Note that this commit also moves loadScript out of the loop so we only call it once. AFAICT, I see no reason to call it on each loop.

### CODE REVIEW

File: app/assets/javascripts/discourse/app/lib/highlight-syntax.js [hunk 1]
```diff
@@ -1,34 +1,51 @@
...
+    elem.querySelectorAll(selector).forEach(e => {
+      // Large code blocks can cause crashes or slowdowns
+      if (e.innerHTML.length > 30000) {
+        return;
+      }
 
-    $(e).removeClass("lang-auto");
-    loadScript(path).then(() => {
+      e.classList.remove("lang-auto");
       customHighlightJSLanguages();
```
Comment (CvX) [inline app/assets/javascripts/discourse/app/lib/highlight-syntax.js:41]: I think this call can also be moved out of the loop?

File: app/assets/javascripts/discourse/app/lib/highlight-syntax.js [hunk 2]
```diff
@@ -1,34 +1,51 @@
...
       hljs.highlightBlock(e);
```
Comment (CvX) [inline app/assets/javascripts/discourse/app/lib/highlight-syntax.js:42]: I don't know how it affects the performance, but before the change it was running all `highlightBlock` concurrently (essentially `forEach(a => new Promise(() => highlightBlock(a)))`) and now it runs them synchronously though still within a promise (i.e. `new Promise(() => forEach(a => highlightBlock(a)))`)

Comment (jjaffeux) [reply app/assets/javascripts/discourse/app/lib/highlight-syntax.js:42]: Hmmm yes indeed, at best it was an unknown desirable side-effect I would say. I doubt anyone would use the promise created by loadScript to make this more performant on purpose.

File: app/assets/javascripts/discourse/app/lib/highlight-syntax.js (hunk 1 above)
Comment (jjaffeux) [reply app/assets/javascripts/discourse/app/lib/highlight-syntax.js:41]: this is on each selected node of the loop not on the top node no?

Comment (CvX) [reply app/assets/javascripts/discourse/app/lib/highlight-syntax.js:41]: I meant `customHighlightJSLanguages` - I don't think the list of languages will change between iterations of the loop. 😉

Comment (jjaffeux) [reply app/assets/javascripts/discourse/app/lib/highlight-syntax.js:41]: Yes I think this is probably something we can pull out of the loop

**Pull request merged**